pkg/install/helm/install.go

```diff
@@ -0,0 +1,147 @@
+package helm
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+
+	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/datawire/dlib/dlog"
+
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+	cl "github.com/telepresenceio/telepresence/v2/pkg/client"
+)
+
+const helmDriver = "secrets"
+const releaseName = "traffic-manager"
+const releaseOwner = "telepresence-cli"
+
+func getHelmConfig(ctx context.Context, configFlags *kates.ConfigFlags, namespace string) (*action.Configuration, error) {
+	helmConfig := &action.Configuration{}
+	err := helmConfig.Init(configFlags, namespace, helmDriver, func(format string, args ...interface{}) {
+		ctx := dlog.WithField(ctx, "source", "helm")
+		dlog.Debugf(ctx, format, args...)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return helmConfig, nil
+}
+
+func getValues(ctx context.Context, clusterID string) map[string]interface{} {
+	clientConfig := client.GetConfig(ctx)
+	imgConfig := clientConfig.Images
+	imageRegistry := imgConfig.Registry
+	imageTag := strings.TrimPrefix(client.Version(), "v")
+	values := map[string]interface{}{
+		"clusterID": clusterID,
+		"image": map[string]interface{}{
+			"registry": imageRegistry,
+			"tag":      imageTag,
+		},
+		"createdBy": releaseOwner,
+	}
+	if mxRecvSize := clientConfig.Grpc.MaxReceiveSize; mxRecvSize != nil {
+		values["grpc"] = map[string]interface{}{
+			"maxReceiveSize": mxRecvSize.String(),
+		}
+	}
+	if imgConfig.WebhookAgentImage != "" {
+		parts := strings.Split(imgConfig.WebhookAgentImage, ":")
+		image := imgConfig.WebhookAgentImage
+		tag := ""
+		if len(parts) > 1 {
+			image = parts[0]
+			tag = parts[1]
+		}
+		values["agentInjector"] = map[string]interface{}{
+			"agentImage": map[string]interface{}{
+				"registry": imgConfig.WebhookRegistry,
+				"name":     image,
+				"tag":      tag,
+			},
+		}
+	}
+
+	return values
+}
+
+func installNew(ctx context.Context, chrt *chart.Chart, helmConfig *action.Configuration, namespace, clusterID string) error {
+	dlog.Info(ctx, "No existing Traffic Manager found, installing...")
+	install := action.NewInstall(helmConfig)
+	install.ReleaseName = releaseName
+	install.Namespace = namespace
+	install.Timeout = 2 * time.Minute
+	install.Atomic = true
+	install.CreateNamespace = true
+	_, err := install.Run(chrt, getValues(ctx, clusterID))
+	return err
+}
+
+func upgradeExisting(ctx context.Context, chrt *chart.Chart, helmConfig *action.Configuration, namespace, clusterID string) error {
+	dlog.Info(ctx, "Existing Traffic Manager found, upgrading...")
+	upgrade := action.NewUpgrade(helmConfig)
+	upgrade.Timeout = 2 * time.Minute
```

Comment:
Nit: Perhaps worth adding a key named something like `helmTimeout` to the timeouts block in config.yml and making this value configurable.. could then use this value for the timeout in `installNew` and `DeleteTrafficManager` if you wanted.